AMI/Components/WebViewCoordinator.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  WebViewCoordinator.swift
+//  AMI
+//
+//  Created by Aline Bonnet on 22/12/2025.
+//
+
+import Foundation
+@preconcurrency import WebKit
+
+class WebViewCoordinator: NSObject, WKNavigationDelegate {
+    var parent: WebView
+    private var urlObservation: NSKeyValueObservation?
+    
+    init(_ parent: WebView) {
+        self.parent = parent
+    }
+    
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        parent.currentURL = navigationAction.request.url!.absoluteString
+        if let urlStr = navigationAction.request.url?.absoluteString, !urlStr.contains(BASE_URL) {
```

Comment:
I'm not sure how `BASE_URL` is available as is, I've tried many different configurations locally, but never managed to make it work, unless I changed it to `Config.shared.BASE_URL`, so there's probably something weird here.

If it works for you and you're able to build, then i guess it's fine for now, but we'll need to fix this at some point so everybody can build 😬